crates/rspack_plugin_javascript/src/visitors/swc_visitor/typescript.rs

```diff
@@ -33,7 +33,6 @@ pub fn typescript<'a>(
         treat_const_enum_as_enum: false,
         ts_enum_is_readonly: assumptions.ts_enum_is_readonly,
       },
-      use_define_for_class_fields: !assumptions.set_public_class_fields,
```

Comment:
removed since it is deprecated by the latest swc. for more details, you could refer https://github.com/swc-project/swc/issues/7085